File: PA62.java
//Leetcode 690. Employee Importance
//Time Complexity : O(n) , n is number of employees
//Space Complexity: O(n) , For hashmap of size equal to no. of employees
class Solution {
    int imp=0;
    HashMap<Integer, Employee> hsub;
    public int getImportance(List<Employee> employees, int id) {
        if(employees ==null || employees.size()==0) return 0;
        hsub= new HashMap<>();
        for(Employee e: employees ){
            hsub.put(e.id,e);           
        }        
        dfs(id);
        return imp;
    }
    private void dfs(int id){
        Employee e= hsub.get(id);
        imp=imp+e.importance;    
        for(Integer i :e.subordinates){
            dfs(i);
        }
        return;
        
    }
}

File: PA_62.py
#LC 690 - Employee Importance
#Time Complexity - O(n)
#Space Complexity - O(n)
"""
# Definition for Employee.
class Employee(object):
    def __init__(self, id, importance, subordinates):
    	#################
        :type id: int
        :type importance: int
        :type subordinates: List[int]
        #################
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""

class Solution(object):
    def getImportance(self, employees, id):
        """
        :type employees: List[Employee]
        :type id: int
        :rtype: int
        """
        adj = dict()
        for emp in employees:
            adj[emp.id] = emp
        self.ans = 0
        def dfs(id):
            self.ans = self.ans + adj[id].importance
            for emp in adj[id].subordinates:
                dfs(emp)
        dfs(id)
        return(self.ans)